src/domain/accounts/domain/src/repos/account_repository.rs

```diff
@@ -21,6 +21,9 @@ use crate::Account;
 pub trait AccountRepository: Send + Sync {
     async fn create_account(&self, account: &Account) -> Result<(), CreateAccountError>;
 
+    // TODO: Private Datasets: tests & stream version
+    async fn get_accounts(&self) -> Result<Vec<Account>, GetAccountsError>;
+
```

Comment:
Overall: having a routine that returns all accounts on the platform is quite an indication of a suboptimal design from performance point of view.

Definitely this should be a stream, rather than a vector. Don't load too much into memory, we recently discussed this.

Also, consider paging from the beginning.

Reply:
A variant of page-by-page loading into the stream is proposed